### PR TITLE
[Console] use STDOUT/ERR in ConsoleOutput to save opening too many file descriptors

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #44882
| License       | MIT
| Doc PR        | -

/cc @jtojnar can you please give this patch a try and report back?

### CODE REVIEW

File: src/Symfony/Component/Console/Output/ConsoleOutput.php [hunk 1]
```diff
@@ -153,14 +153,18 @@ private function openOutputStream()
...
     }
 
     /**
      * @return resource
      */
     private function openErrorStream()
     {
-        return fopen($this->hasStderrSupport() ? 'php://stderr' : 'php://output', 'w');
+        if (!$this->hasStderrSupport()) {
+            return fopen('php://output', 'w');
```
Comment (derrabus) [inline src/Symfony/Component/Console/Output/ConsoleOutput.php:166]: Shall we use `STDOUT` here as well if it's defined?

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Console/Output/ConsoleOutput.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Console/Output/ConsoleOutput.php:166]: php://output !== php://stdout, so not sure

Comment (derrabus) [reply src/Symfony/Component/Console/Output/ConsoleOutput.php:166]: True. 🤔

File: src/Symfony/Component/Console/Output/ConsoleOutput.php [hunk 2]
```diff
@@ -153,14 +153,18 @@ private function openOutputStream()
             return fopen('php://output', 'w');
         }
 
-        return @fopen('php://stdout', 'w') ?: fopen('php://output', 'w');
+        return \defined('STDOUT') ? \STDOUT : (@fopen('php://stdout', 'w') ?: fopen('php://output', 'w'));
```
Comment (chalasr) [inline src/Symfony/Component/Console/Output/ConsoleOutput.php:157]: in absence of test case, a small comment would be nice in case this does fix the issue

Comment (nicolas-grekas) [reply src/Symfony/Component/Console/Output/ConsoleOutput.php:157]: comment added

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Comment by jtojnar** [comment]
Can confirm this fixes the issue for me. Thanks.

**Comment by TomasVotruba** [comment]
Thank you :+1: it fixed one issue we have with Rector tests

**Pull request merged**